app/actions/index.js

```diff
@@ -125,7 +168,7 @@ export const clearRemoteSurveys = () => (dispatch, getState) =>
     type: types.CLEAR_REMOTE_SURVEYS
   });
 
-export const fetchRemoteSurvey = (id, url) => (dispatch, getState) => {
+export const fetchRemoteSurvey = (id, url, target) => (dispatch, getState) => {
```

Comment:
I'm a little uncomfortable with overloading functionality here. That might be overcome slightly by changing the name of this function, but it's more about fetching multiple surveys and triggering data synchronization each time.